mongo_connector/dest_mapping.py

```diff
@@ -0,0 +1,186 @@
+        num_match = src_match.lastindex
+        if num_match != map_pattern.count("*"):
+            LOG.error("The number of wildcards in the mapping %s should be %d" %(map_pattern, num_match))
+            return None
+        pattern_arr = re.split('(\*)', map_pattern)
+        map_arr = []
+        map_index = 1
+        for x in pattern_arr:
+            if x != "*":
+                map_arr.append(x)
+            else:
+                map_arr.append(src_match.group(map_index))
+                map_index += 1
+        
+        return ''.join(map_arr)
+        
+    
+    def get(self,plain_src_ns,defaultval=""):
+        """Given a plain source namespace, return a mapped namespace if matched.
+        If no match and given defaultval, return it as a default value.
+        """
+        self.lock.acquire()
+        # search in plain mappings first
+        if plain_src_ns in self.plain.keys():
+            self.lock.release()
+            return self.plain[plain_src_ns]
+        else:
+            # search in wildcard mappings
+            # if matched, get a replaced mapped namespace
+            # and add to the plain mappings            
+            for k, v in self.wildcard.items():
+                m_res = self.match(plain_src_ns, k)
+                if m_res:
+                    res = self.replace(m_res, v)
+                    if res is not None:
+                        if res not in self.reverse_plain:
+                            self.plain[plain_src_ns] = res
+                            self.reverse_plain[res] = plain_src_ns
+                            
+                            # add matched plain db to plain_db dict, but skip $cmd case
+                            db, col = plain_src_ns.split(".", 1)
+                            if col != "$cmd":
+                                arr = self.plain_db.get(db, [])
+                                arr.append(res.split(".")[0])
+                                self.plain_db[db] = list(set(arr))
+                            self.lock.release()
+                            return res
+                        else:
+                            self.lock.release()
+                            raise errors.InvalidConfiguration("Destination namespaces set wildcard pattern conflicts with the plain pattern.")
+            
+        if defaultval:
+            self.lock.release()
+            return defaultval
+        else:
+            self.lock.release()
+            LOG.error("Failed to find matched mapping.")
+            return None
+    
+
+    def set(self, key, value):
+        """Set the DestMapping instance to corresponding dictionary"""
+        self.lock.acquire()
+        db = key.split(".")[0]
+        if "*" in db:
+            raise errors.InvalidConfiguration("Wildcard is not allowed in db name")
+            
+        if "*" in key:
+            if key.count("*") != value.count("*"):
```

Comment:
I think we should limit the number of `*` characters in any pattern to 1. Any more than that, and this feature gets very complicated. Does your use case require being able to match more than one wildcard character in a namespace?